### PR TITLE
Orbit failsafe: Prevent crashes because of zero thrust 

### PR DESCRIPTION
**Pull request by Stifael**
**Issue**
In contrast to all tasks within Flighttasks, the orbit-task can only be invoked via vehicle_command msg, which is the uorb equivalent of the mavlink `MAV_CMD`-msg. The orbit-task currently cannot be activated solely via the usual commander-state -> navigation-state pattern, but rather requires a vehicle_command msg to be sent and received in Flighttasks, which then starts the orbit-task. During normal operation, that does not cause any issue. However, during failsafe (for instance RC-loss), the flight-mode changes from orbit into another failsafe mode (for instance RTL) and if there is a recovery (RC regained), then the commander wants to switch into orbit again, which internally also sets the `vehicle_status.nav_state` to `NAVIGATION_STATE_ORBIT`. Within `mc_pos_control`, the NAVIGATION_STATE_ORBIT will not ensure that Flighttasks will switch into FlightTaskOrbit (https://github.com/PX4/Firmware/blob/master/src/modules/mc_pos_control/mc_pos_control_main.cpp#L900-L901), but rather assumes that the vehicle_command will do that (https://github.com/PX4/Firmware/blob/master/src/lib/FlightTasks/FlightTasks.cpp#L152), which will never happen because during failsafe recovery no vehicle_command msg is published.  

This issue is quite serious because depending on the failsafe mechanism, the drone can be stuck in FlighttaskAuto while the rest of the system thinks it is back in orbit. Furthermore, if RC loss is only for a few loops, it can happen that FlighttaskAuto gets initialized with old triplets which can lead to flyaways or almost freefall. Both happened on a real vehicle and can be simulated via HITL (I think SITL could also work but requires different changes in the code).

**Solution**
During Orbit, switch internal-main state from Orbit to PositionCtl if failsafe is triggered. 

**Test data / coverage**
It was observed on a real vehicle  (the vehicle fell from the sky after RC loss-regain during orbit). Thanks to @potaito, the bug could be reproduced in HITL by generating fake rc-loss. In principle, using this branch https://github.com/PX4/Firmware/commits/px4/hotfix/debug/orbit-failsafe-recovery-into-posctl, the error can be reproduced.

**Describe your preferred solution**
Commander refactor  :laughing:

### CODE REVIEW

File: src/modules/commander/state_machine_helper.cpp [hunk 1]
```diff
@@ -593,22 +593,39 @@ bool set_nav_state(vehicle_status_s *status, actuator_armed_s *armed, commander_
 			// failsafe: on engine failure
 			status->nav_state = vehicle_status_s::NAVIGATION_STATE_AUTO_LANDENGFAIL;
 
+			// Orbit can only be started via vehicle_command (mavlink). Recovery from failsafe into orbit
+			// is not possible and therefore the internal_state needs to be adjusted.
+			internal_state->main_state = commander_state_s::MAIN_STATE_POSCTL;
+
 		} else if (is_armed && check_invalid_pos_nav_state(status, old_failsafe, mavlink_log_pub, status_flags, false, true)) {
-			// failsafe: necessary position estimate lost (nothing to do - everything done in check_invalid_pos_nav_state)
+			// failsafe: necessary position estimate lost; witching is done in check_invalid_pos_nav_state
```
Comment (potaito) [inline src/modules/commander/state_machine_helper.cpp]: typo in `witching` :p

Comment (bresch) [reply src/modules/commander/state_machine_helper.cpp]:
```suggestion
			// failsafe: necessary position estimate lost; switching is done in check_invalid_pos_nav_state
```

**Comment by MaEtUgR** [comment]
@Stifael Thanks for fixing this. IMHO this shows again how broken the state machine really is. There are so many cases in which if you don't put the right if statement somewhere in the commander the outcome is non-deterministic. This fix totally makes sense. Sorry I didn't see that coming, I remember trying mode switches and failsafes when setting up how orbit is triggered but probably I tested the wrong cases. I hope we get to a level where such safety critical issues are handled implicitly for any mode.

**Comment by potaito** [comment]
@MaEtUgR I agree with our assessment. And don't worry, mistakes happen. But I would still like to point out how critical this bugfix is, since without it drones can run into a corner case where they just drop in freefall :sweat_smile: I'm glad we were able to find it before a serious crash happened.

**Comment by MaEtUgR** [comment]
@potaito Yes, I totally understood. What I'd like to investigate is how it even got into free fall. Even in this weird corner case if there's no proper setpoint the controller should resort to an unexpected but safe hover followed by a landing.

**Comment by potaito** [comment]
Yes please, if you can find out what's going on it would be much appreciated :see_no_evil:

**Comment by Stifael** [comment]
> What I'd like to investigate is how it even got into free fall

I think that is quite clear now and I tried to explain it above :see_no_evil: 

>  if RC loss is only for a few loops, it can happen that FlighttaskAuto gets initialized with old triplets which can lead to flyaways or almost freefall.

This basically means that FlightTasks tell the mc_pos_controller to do that because of wrong initialization. 

> IMHO this issue shows again how broken the state machine really is

I think we all agree on that too. But to be fair, the orbit is quite some special case with a state-machine that differs from the other.

**Pull request merged**